python/ray/serve/handle.py

```diff
@@ -60,17 +115,14 @@ def remote(self, request_data: Optional[Union[Dict, Any]] = None,
             ``**kwargs``: All keyword arguments will be available in
                 ``request.args``.
         """
-        request_metadata = RequestMetadata(
-            get_random_letters(10),  # Used for debugging.
-            self.endpoint_name,
-            TaskContext.Python,
-            call_method=self.method_name or "__call__",
-            shard_key=self.shard_key,
-            http_method=self.http_method or "GET",
-            http_headers=self.http_headers or dict(),
-        )
-        return self.router_handle.enqueue_request.remote(
-            request_metadata, request_data, **kwargs)
+        assert self.sync, "handle.remote() should be called from sync handle."
+        coro = self._remote(request_data, kwargs)
+        future = asyncio.run_coroutine_threadsafe(coro, self.async_loop)
+        return future.result()
+
+    async def remote_async(self, request_data, **kwargs):
```

Comment:
please mark this experimental for now so that we can change the API in the near future if we want to